src/Microsoft.TemplateEngine.Core/Util/CombinedStream.cs

```diff
@@ -38,26 +38,45 @@ public override void Flush()
 
         public override int Read(byte[] buffer, int offset, int count)
         {
-            int read = 0;
-            if (!_isStream1Depleted)
+            if (count + offset > buffer.Length)
             {
-                read = _stream1.Read(buffer, offset, count);
+                //cannot read more than available buffer length
+                count = buffer.Length - offset;
+            }
 
-                if (read == count)
+            int totalBytesRead = 0;
+            if (!_isStream1Depleted)
+            {
+                while (totalBytesRead < count)
```

Comment:
I suggest we do partial reads here and leave handling this to the caller, just like any other streams.

Reply:
let's delay it when tests on InlineMarkupConditional (where CombinedStream is used) are written. I prefer to have keep implementation as is for now.